src/Factory/Address/AddressViewFactory.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusLagersystemPlugin\Factory\Address;
+
+use Setono\SyliusLagersystemPlugin\View\AddressView;
+use Sylius\Component\Core\Model\AddressInterface;
+
+final class AddressViewFactory implements AddressViewFactoryInterface
+{
+    /** @var string */
+    private $addressViewClass;
+
+    public function __construct(string $addressViewClass)
+    {
+        $this->addressViewClass = $addressViewClass;
+    }
+
+    public function create(AddressInterface $address): AddressView
+    {
+        /** @var AddressView $addressView */
+        $addressView = new $this->addressViewClass();
```

Comment:
I am not sure, but maybe in this case I would create a method like:

```
private function createView(): AddressView
{
    return new $this->addressViewClass;
}
```

This way you would have the type hint to enforce the type earlier and you wouldn't have to add a `@var` annotation.

Reply:
I've used standard approach from ShopAPI. I believe its better to follow it so any dev who experienced with ShopAPI can understand easily what we have in our plugin.